### PR TITLE
Rfe trello 2152 refactor ufg flow

### PR DESCRIPTION
**Pull request by talgoldy**

### CODE REVIEW

File: eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/EyesService.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 public class EyesService extends Thread {
 
-    protected final int threadPoolSize;
+    protected final int testsPoolSize;
```
Comment (talgoldy) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/EyesService.java:11]: eyesConcurrency

File: eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/RunningTest.java [hunk 2]
```diff
@@ -171,8 +170,8 @@ public ScoreTask getScoreTaskObjectByType(VisualGridTask.TaskType taskType) {
         if (!this.isTestOpen.get() && taskType == VisualGridTask.TaskType.CHECK) {
             return null;
         }
+
         int score = 0;
-        VisualGridTask chosenVisualGridTask;
         synchronized (this.visualGridTaskList) {
             for (VisualGridTask visualGridTask : this.visualGridTaskList) {
                 if (visualGridTask.isTaskReadyToCheck() && visualGridTask.getType() == VisualGridTask.TaskType.CHECK) {
```
Comment (talgoldy) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/RunningTest.java:177]: change to ready to render

File: eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/VisualGridTask.java [hunk 3]
```diff
@@ -103,17 +102,15 @@ public TestResultContainer call() {
...
-                        RectangleSize deviceSize = renderResult.getDeviceSize();
-                        eyesConnector.setUserAgent(userAgent);
-                        eyesConnector.setDeviceSize(deviceSize);
-                    } else {
-                        // We are in exception mode - trying to do eyes.open() without first render
-                        RenderBrowserInfo browserInfo = runningTest.getBrowserInfo();
-                        //eyesConnector.setUserAgent(craftUserAgent(browserInfo));
-                        eyesConnector.setDeviceSize(browserInfo.getViewportSize());
+                    String userAgent = getUserAgent();
+                    RectangleSize deviceSize = getCorrectDeviceSize();
```
Comment (talgoldy) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/VisualGridTask.java:106]: change method name

File: eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/VisualGridTask.java [hunk 4]
```diff
@@ -167,6 +158,9 @@ public TestResultContainer call() {
 
                 case ABORT:
                     logger.verbose("VisualGridTask.run abort task");
+                    if (!runningTest.isTestOpen()) {
+                        return new TestResultContainer(testResults, runningTest.getBrowserInfo(), this.exception);
+                    }
```
Comment (itaibh) [inline eyes.sdk.core/src/main/java/com/applitools/eyes/visualgrid/services/VisualGridTask.java:163]: Why not reverse the `if` and let the default operation return just that?
A side effect will be notifying all the listeners, but isn't it a wanted side effect?

**Pull request merged**